src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -126,15 +183,16 @@ protected override async Task ExecuteAsync(EventContext telemetryContext, Cancel
                 string[] viewsetArray = this.Viewset.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries);
                 string specviewOriginalLogPath;
                 string specviewRenamedLogPath;
-                List<Task> tasks = new List<Task>();
+                var tasks = new List<Task>();
+
                 foreach (string viewset in viewsetArray)
                 {
                     // log file is inside a folder that starts with the viewset name e.g. 3dsmax-07
-                    string? viewsetLogDir = this.fileSystem.Directory.GetDirectories(this.historyResultsPath, $"{viewset}*", SearchOption.TopDirectoryOnly).FirstOrDefault();
+                    string? viewsetLogDir = this.PlatformSpecifics.Combine(this.historyResultsPath, this.viewsetLogFileNameMapping[viewset]);
                     if (viewsetLogDir == null)
                     {
                         throw new WorkloadResultsException(
-                            $"The expected SPECviewperf viewset log directory was not found in '{this.historyResultsPath}'.",
+                            $"The expected SPECviewperf viewset directory {viewset} was not found in '{this.historyResultsPath}'.",
                             ErrorReason.WorkloadResultsNotFound);
                     }
 
```

Comment:
Let's not parallelize the file upload with tasks inside of the executor. You can have a List<> of files you want to upload for all viewsets, and then upload the whole list in one call.